packages/core/src/components/Link/Link.tsx

```diff
@@ -19,7 +19,7 @@ import { Link as MaterialLink } from '@material-ui/core';
 import { Link as RouterLink } from 'react-router-dom';
 
 type Props = ComponentProps<typeof MaterialLink> &
-  ComponentProps<typeof RouterLink> & { component?: React.FC<any> };
+  ComponentProps<typeof RouterLink> & { component?: React.ReactNode };
```

Comment:
Realized `component` here should be a component or tag name, see https://material-ui.com/api/link/.

The usage of `FC` here was actually not the typical and forbidden one 😅 , but I think it still makes sense to switch to `ComponentType<{}> | string`

Reply:
yeah, makes sense. 
One question, is this component designed to receive only react-router-dom component as a prop, or can it receive some other component as well.

The reason for the question, we can't just use `ComponentType<{}>`, since it will mean it receives a component without any props (except for `children`), and the way it is used now, it passes the Link from react router, that has prop `to`

If this component only receives react router dom link, I can put 
`ComponentType<LinkProps> | string` and LinkProps here is exported pros for link from react-router-dom
or use any, if any component can be passed
`ComponentType<any> | string`

Reply:
@Rugvip, what do you think? I would like to merge this, before there are more conflicts 😄

Reply:
@magneticz ended up fixing this a bit swiftly in #3665, since release is about to happen 😅